src/Dhall/JSON.hs

```diff
@@ -317,15 +317,14 @@ dhallToJSON e0 = loop (Dhall.Core.normalize e0)
                     loop (Dhall.Core.RecordLit contents')
                 _ -> do
                     a' <- traverse loop a
-                    return (Data.Aeson.toJSON a')
+                    return (Data.Aeson.toJSON (Dhall.Map.toMap a'))
         Dhall.Core.UnionLit _ b _ -> loop b
         _ -> Left (Unsupported e)
 
-toOrderedList :: Ord k => InsOrdHashMap k v -> [(k, v)]
+toOrderedList :: Ord k => Dhall.Map.Map k v -> [(k, v)]
 toOrderedList =
         Data.List.sortBy (Data.Ord.comparing fst)
-    .   Data.HashMap.Strict.toList
-    .   Data.HashMap.Strict.InsOrd.toHashMap
+    .   Dhall.Map.toList
```

Comment:
We know that now `Dhall.Map.toList` is already sorted, dont sure if we should remove the extra sorting (cause we dont know for sure that it will sorted with another impl).
If `Dhall.Map` would add an `toOrderedList` to its api we could use it safely.